src/Command/BuildDocsCommand.php

```diff
@@ -129,7 +129,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         foreach ($this->finder as $file) {
             $htmlFile = str_replace([$this->sourceDir, '.rst'], [$this->htmlOutputDir, '.html'], $file->getRealPath());
-            if (!$this->filesystem->exists($htmlFile)) {
+
+            $firstLine = fgets(fopen($file->getRealPath(), 'r'));
+
+            if (!$this->filesystem->exists($htmlFile) && ':orphan:' !== trim($firstLine)) {
```

Comment:
When we refactor the logic of this command into a service, we could update our functional test to check for this - i.e. have 2 files that aren't rendred, but make 1 an orphan, and check that we correctly see 1 unrendered file.

Also, this is *exactly* the kind of stuff that we should not need to do :). We should be able to get the "meta" for this document and check if it is an "orphan". If we're missing that from the meta (I think you said this was the case), we should open an issue on the rst-parser repo.

Reply:
https://github.com/doctrine/rst-parser/issues/88
;)